proof/refine/ARM_HYP/KHeap_R.thy

```diff
@@ -258,13 +256,6 @@ lemma updateObject_default_result:
   "(x, s'') \<in> fst (updateObject_default e ko p q n s) \<Longrightarrow> x = injectKO e"
   by (clarsimp simp add: updateObject_default_def in_monad)
 
-lemma ps_clear_upd':
-  "ksPSpace s y = Some v \<Longrightarrow>
-    ps_clear x n (s' \<lparr> ksPSpace := ksPSpace s(y \<mapsto> v')\<rparr>) = ps_clear x n s"
-  by (rule iffI | clarsimp elim!: ps_clear_domE | fastforce)+
-
-lemmas ps_clear_updE'[elim] = iffD2[OF ps_clear_upd', rotated]
-
```

Comment:
This is interesting. So, in ARM_HYP we somehow have exactly the same thing defined as `ps_clear_upd`, and this is a duplicate, which you removed. I’d actually prefer not having ‘ at the end in other arches, but I guess this is not a high priority thing. Maybe one day :)

Reply:
Yes, that should probably be a separate thing. I'm not sure any more why I noticed the duplication here, it's possible that Isabelle complained about something or gave a warning.